tests/TestCase/Error/ExceptionRendererTest.php

```
@@ -731,23 +731,23 @@ public function testRenderExceptionInBeforeRender()
      */
     public function testMissingLayoutPathRenderSafe()
     {
+        $eventTriggered = false;
         $exception = new NotFoundException();
         $ExceptionRenderer = new ExceptionRenderer($exception);
 
-        $ExceptionRenderer->controller = $this->getMockBuilder('Cake\Controller\Controller')
-            ->setMethods(['render'])
-            ->getMock();
+        $ExceptionRenderer->controller = new Controller();
         $ExceptionRenderer->controller->helpers = ['Fail', 'Boom'];
-        $ExceptionRenderer->controller->eventManager()->on('Controller.beforeRender', function (Event $event) {
-            $event->subject()->viewBuilder()->layoutPath('boom');
-        });
+        $ExceptionRenderer->controller->eventManager()->on(
+            'Controller.beforeRender',
+            function (Event $event) use ($exception, &$eventTriggered) {
+                $eventTriggered = true;
+                $event->subject()->viewBuilder()->layoutPath('boom');
+
+                throw $exception;
```

Do you need to throw here, or will the invalid layout path cause an error?